subcommand/injector/command.go

```diff
@@ -231,6 +232,7 @@ func (c *Command) Run(args []string) int {
 
 	// Registering path to expose metrics
 	if c.flagTelemetryPath != "" {
+		agentInject.MustRegisterInjectorMetrics(prometheus.DefaultRegisterer)
```

Comment:
Note that this ultimately calls `registry.MustRegister`, which will [panic](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#MustRegister) on failure. The behavior matches [what's done in VSO](https://github.com/hashicorp/vault-secrets-operator/blob/1273434f78c5a2561ec461715306d146cc13fe69/main.go#L366) but wanted to call it out just in case.